### PR TITLE
fix(fix_xception):fix xception & start fix loss func

### PR DESCRIPTION
**Pull request by teamwong111**
首先说一下，这次push主要是想本地修一下loss
然后关于修改xception的过程：
1. 首先发现在初始conv没有加input_shape
2. 分别测试每个模块文件，发现并没有问题，这个阶段没有进行完备性测试
3. 因此肯定是因为有些模块传入参数的不同导致问题，所以针对不同的参数进行测试，发现是因为有些条件性的self.xxx没有用if包裹

### CODE REVIEW

File: test/test_train/test_train_routine.py [hunk 1]
```diff
@@ -45,27 +45,35 @@ def test_run(self):
 
         self.assertTrue(os.path.exists('exp/config/exp_config.json'))
 
-    def diff_model(self):
-        ds = get_tr_ds(original_pattern='/opt/dataset/tr2_cropped/data/*.png',
-                       mask_pattern='/opt/dataset/tr2_cropped/label/*.png',
-                       batch_size=16)
+    def test_diff_model(self):
```
Comment (chong1144) [inline test/test_train/test_train_routine.py]: 这块改一下，不然 CI 过不了

File: segelectri/loss_metrics/loss.py [hunk 2]
```diff
@@ -14,7 +14,15 @@ def __init__(self, gamma=2.0, alpha=0.25, **kwargs):
...
     def call(self, y_true, y_pred):
+        y_true = tf.cast(y_true[:, :, :, 1], dtype=tf.int32)
+        y_pred = tf.cast(y_pred, dtype=tf.float32)
+        print("----------------------------")
+        print(tf.shape(y_pred))
+        print(y_pred.shape)
         y_true = tf.one_hot(y_true, y_pred.shape[3])
+        print(tf.shape(y_true))
+        print(y_true.shape)
+
```
Comment (chong1144) [inline segelectri/loss_metrics/loss.py:25]: 这些 print 用 tf 自身的 logging 更好一些？

File: segelectri/model/deeplab/__init__.py [hunk 3]
```diff
@@ -14,34 +14,38 @@ def __init__(self,
...
                  **kwargs):
-        super().__init__(*args, **kwargs)
+        super(Deeplab, self).__init__(*args, **kwargs)
 
         self.dilation_rates = dilation_rates
         self.num_classes = num_classes
 
-        self.xception = Xception()
+        # self.xception = Xception()
+        self.mobilenet = Mobilenet()
```
Comment (chong1144) [inline segelectri/model/deeplab/__init__.py:23]: 这块不要用注释吧，用 backbone 的方式来做一个映射？就是 backbone 是一个字符串，比如是 'xception' | 'mobilenet' 这样

File: segelectri/model/deeplab/mobilenet/backbones.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    'resnet50': {
+        'model': tf.keras.applications.ResNet50,
+        'feature_1': 'conv4_block6_2_relu',
+        'feature_2': 'conv2_block3_2_relu'
+    },
+    'mobilenetv2': {
+        'model': tf.keras.applications.MobileNetV2,
+        'feature_1': 'block_3_expand_relu',
+        'feature_2': 'block_13_expand_relu'
+    }
```
Comment (chong1144) [inline segelectri/model/deeplab/mobilenet/backbones.py:13]: 这个 backbones 既有 resnet 还有 moblienet 但是竟然在 mobilenet 这个文件夹下，感觉有些怪

File: test/test_model/deeplab/test_deeplab_layers.py [hunk 5]
```diff
@@ -27,13 +27,18 @@ def test_aspp_with_real_input(self):
...
+        self.assertAllEqual(outputs.shape, [None, 1024, 1024, 6])
 
         decoder_serial = decoder.get_config()
         self.assertTrue(decoder_serial['num_classes'], 6)
+
+
+if __name__ == '__main__':
+    test = TestDeeplabLayers()
+    test.test_aspp_with_real_input()
+    test.test_decoder()
```
Comment (chong1144) [inline test/test_model/deeplab/test_deeplab_layers.py:44]: tf.test.main() 这样吧

File: test/test_model/deeplab/test_mobilenet.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        mobile_serial = mobile.get_config()
+        print(mobile_serial)
+        self.assertAllEqual(outputs.shape, [2, 64, 64, 576])
+        self.assertAllEqual(skip.shape, [2, 256, 256, 144])
+
+
+if __name__ == '__main__':
+    test = TestDeeplabLayers()
+    test.test_mobilenet_with_real_input()
+    # tf.test.main()
```
Comment (chong1144) [inline test/test_model/deeplab/test_mobilenet.py:35]: 同 tf.test.main()

File: test/test_train/test_train_routine.py [hunk 7]
```diff
@@ -45,27 +45,35 @@ def test_run(self):
...
+        routine.run(exp_dir='exp/03', epochs=1)
 
-        self.assertTrue(os.path.exists('exp/01/config/exp_config.json'))
+        self.assertTrue(os.path.exists('exp/03/config/exp_config.json'))
 
 
 if __name__ == '__main__':
-    unittest.main()
+    TestTrainRoutine().test_diff_model()
+    # unittest.main()
```
Comment (chong1144) [inline test/test_train/test_train_routine.py]: 同 tf.test.main()

File: test/test_train/test_train_routine.py (hunk 1 above)
Comment (teamwong111) [reply test/test_train/test_train_routine.py]: 这里改了，然后发现test_loss那里没改。。。

File: segelectri/loss_metrics/loss.py (hunk 2 above)
Comment (teamwong111) [reply segelectri/loss_metrics/loss.py:25]: tf.logging好像是tf1.0的，我用tf.print吧

File: segelectri/model/deeplab/__init__.py (hunk 3 above)
Comment (teamwong111) [reply segelectri/model/deeplab/__init__.py:23]: 行

File: segelectri/model/deeplab/mobilenet/backbones.py (hunk 4 above)
Comment (teamwong111) [reply segelectri/model/deeplab/mobilenet/backbones.py:13]: 行，我给它提出来

File: test/test_model/deeplab/test_deeplab_layers.py (hunk 5 above)
Comment (teamwong111) [reply test/test_model/deeplab/test_deeplab_layers.py:44]: 好的

File: segelectri/loss_metrics/loss.py (hunk 2 above)
Comment (chong1144) [reply segelectri/loss_metrics/loss.py]: 就用标准的 logging 库吧，好像 tf logging 是兼容的

Comment (chong1144) [reply segelectri/loss_metrics/loss.py]: 然后输出的格式规范一下吧
eg.
``` python
logging.debug('shape of y_true is %s', tf.shape(y_true))
```
logging 的级别你可以来定，但是输出可读高一些

**Review by chong1144 — APPROVED** [review]
LGTM
